### PR TITLE
RDM-3479: Definitive Case Type ID check on Definition import

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-3479.

### Change description ###
Check that capitalisation of Case Type IDs matches existing ones in the Definition Store, blocking the import if this is not the case.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeService.java [hunk 1]
```diff
@@ -20,4 +20,5 @@
 
     Optional<CaseTypeVersionInformation> findVersionInfoByCaseTypeId(String id);
 
+    String findDefinitiveCaseTypeId(String id);
 }
```
Comment (adr1ancho) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeService.java]: Is this a findBy method?  Not entirely sure the purpose of the method yet but I shall read on.

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/exception/SpreadsheetValidationErrorMessageCreator.java [hunk 2]
```diff
@@ -449,6 +449,24 @@ public String createErrorMessage(EventCaseFieldMetadataValidatorImpl.ValidationE
...
+    }
+
+    @Override
+    public String createErrorMessage(CaseTypeEntityReferenceSpellingValidationError error) {
+        return newMessageIfDefinitionExists(error,
+            error.getCaseTypeEntity(),
+            def -> String.format(
+                "Case Type with ID '%s' on tab '%s' already exists with the definitive spelling '%s'. "
+                    + "This spelling must be used for the Case Type ID.",
+                error.getCaseTypeEntity().getReference(),
```
Comment (adr1ancho) [inline excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/exception/SpreadsheetValidationErrorMessageCreator.java]: A little awkward to see String.format and a plus sign in the same statement

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeService.java (hunk 1 above)
Comment (Daniel-Lam) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeService.java]: @adr1ancho - the implementation class calls `findFirstByReferenceIgnoreCaseOrderByCreatedAtDescIdDesc()` in `CaseTypeRepository`.

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/exception/SpreadsheetValidationErrorMessageCreator.java (hunk 2 above)
Comment (adr1ancho) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/exception/SpreadsheetValidationErrorMessageCreator.java]: If we are not careful, we can end up with line like https://github.com/hmcts/ccd-definition-store-api/blob/16c0f885056b45141bf4f64164a3c57ffe1460a3/excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/exception/SpreadsheetValidationErrorMessageCreator.java#L116


A suggestion is 

String.format(
                "Case Type with ID '%s' on tab '%s' already exists with the definitive spelling '%s'. %s",
                error.getCaseTypeEntity().getReference(),
                def.getSheetName(),
                error.getDefinitiveCaseTypeReference()),
                "This spelling must be used for the Case Type ID."

**Pull request merged**